sandbox/libs/authn/src/main/java/org/opensearch/authn/Permission.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+/**
+ * This is an abstract class that defines the minimum expectations for a permission object.
+ * A permission implementation of this class can implement further functionality or structuring.
+ * A permission needs to entail the action it allows and the resource its performed on.
+ */
+abstract class Permission {
+
+    // If using a string for construction, a delimiter is required to split the string
+    String PERMISSION_DELIMITER;
+
+    // If using string-object permissions, you use the invalid characters for ensuring formatting
+    String[] INVALID_CHARACTERS;
+
+    // An array of the valid operations which a permission can grant the privilege to perform.
+    String[] QUALIFIED_OPERATIONS;
+
+    // An array of the available resources which a permission can grant some operation to act upon.
+    String[] QUALIFIED_RESOURCES;
```

Comment:
I really like the notion of having required qualifications, how do you think we can handle enforcing permissions that do not follow this schema?

Reply:
For enforcing/validating the structure of permissions, I was thinking we would need to make it so that each schema has either a translator to interact with a common API or has their own equivalent. I am not really sure what would be helpful for permissions other strings but I am sure someone would have a use. I think this can be made more general once the parts (see below about the number of parts in the permissions) is decided.